From c866a2f8603b44e89eb21a6cf7d88134af2e8b66 Mon Sep 17 00:00:00 2001
From: Bin Liu <bliu@suse.com>
Date: Fri, 10 Mar 2017 15:22:13 +0800
Subject: [PATCH] logconfig: Do not overwrite logger_subsys priority

logfile_priority and syslog_priority could be modified by
logging.logger_subsys.{logfile_priority|syslog_priority}. which could
lead to the following output(which are at notice level):

corosync[21419]:   [QUORUM] Using quorum provider corosync_votequorum
corosync[21419]:   [QUORUM] Members[1]: 1084777643
corosync[21419]:   [QUORUM] This node is within the primary component
                   and will provide service.
corosync[21419]:   [QUORUM] Members[3]: 1084777563 1084777584 1084777643

even the syslog_priority is warning. This patch could avoid the
overwrite.

Signed-off-by: Bin Liu <bliu@suse.com>
Reviewed-by: Jan Friesse <jfriesse@redhat.com>
(cherry picked from commit d2a5e1442e970069d335ce3a8a67c3b9af352a34)
---
 exec/logconfig.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/exec/logconfig.c b/exec/logconfig.c
index a4d348a..6d0bed6 100644
--- a/exec/logconfig.c
+++ b/exec/logconfig.c
@@ -401,7 +401,7 @@ static int corosync_main_config_set (
 			goto parse_error;
 		}
 	}
-	else {
+	else if(strcmp(key_name, "logging.syslog_priority") == 0){
 		if (logsys_config_syslog_priority_set(subsys,
 						      logsys_priority_id_get("info")) < 0) {
 			error_reason = "unable to set syslog level";
@@ -449,7 +449,7 @@ static int corosync_main_config_set (
 			goto parse_error;
 		}
 	}
-	else {
+	else if(strcmp(key_name,"logging.logfile_priority") == 0){
 		if (logsys_config_logfile_priority_set(subsys,
 						      logsys_priority_id_get("info")) < 0) {
 			error_reason = "unable to set syslog level";
-- 
1.7.1

